.github/workflows/ci.yml

```diff
@@ -63,16 +63,16 @@ jobs:
 
           # Ubuntu sub-jobs:
           # ================
-          # GCC 11
+          # GCC 11 (with latest language standards)
```

Comment:
"latest" is a tricky word here - C23/C++23 are effectively complete and just going through the ISO publication process[^1]. Both GCC/Clang support `-std=c2x` and `-std=c++2b` for a while already, and have decent implementation coverage. Once the final standards are published the compilers just add another alias to the existing flags, and more importantly, these flags get used in production projects long before the official flags are available in a release (presumably GCC14, and maybe a GCC13 point release backport; note that GCC 12 is already out).

It's possible that these upcoming standards are too much of a moving target for cython, but if the intention is to catch breakages early, then I think it's actually worth considering to go with the newest ones.

[^1]: yes, standardization national bodies are still able to raise comments that might lead to small changes, but those are hardly ever big changes.

Reply:
Yeah agreed but...

The issue that prompted this was mostly C++14/17-level features that we were using but weren't testing (`[[maybe_unused]]`). I'm not too worried about locking to the absolute latest level of standard until we start using anything from it.

So I think C++20 is good enough to test what I want for the moment. It's more about making sure that when we conditionally use features from recent standards then they get tested, rather than worrying about the standard break Cython.